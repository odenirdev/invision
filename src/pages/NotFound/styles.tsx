import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: row-reverse;

  width: 100%;
  height: 90vh;

  @media (max-width: 750px) {
    flex-direction: column-reverse;
    align-items: center;
    justify-content: center;
  }

  header {
    width: 50%;

    display: flex;
    flex-direction: column;
    justify-content: center;

    h1 {
      font-size: 7vh;

      @media (max-width: 750px) {
        font-size: 4vh;
      }

      span {
        font-size: 10vh;
      }
    }

    p {
      margin-top: 5vh;
      font-size: 3vh;
    }

    button {
      margin-top: 5vh;
    }
  }

  img {
    max-width: 700px;

    margin: 5vh;

    @media (max-width: 750px) {
      height: 25%;
      margin: 0;
    }
  }
`;
