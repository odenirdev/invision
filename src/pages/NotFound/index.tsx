import React from "react";
import { useHistory } from "react-router-dom";

import { Container } from "./styles";

import Button from "../../components/Button";
import Navbar from "../../components/Navbar";

import ContructionIllustration from "../../assets/images/contruction-illustration.svg";

const NotFound: React.FC = () => {
  const history = useHistory();

  return (
    <>
      <Navbar />
      <Container>
        <header>
          <h1>
            <span>404</span> <br />
            Page Not Found
          </h1>
          <p>Page in development, come back soon</p>

          <Button
            backgroundColor="var(--primary-color)"
            color="#FFF"
            onClick={() => history.push("/")}
          >
            Go to home
          </Button>
        </header>
        <img src={ContructionIllustration} alt="Developing page" />
      </Container>
    </>
  );
};

export default NotFound;
