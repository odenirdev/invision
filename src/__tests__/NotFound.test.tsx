import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import NotFound from "../pages/NotFound";

it("renders", () => {
  const { asFragment } = render(<NotFound />);

  expect(asFragment()).toMatchSnapshot();
});
