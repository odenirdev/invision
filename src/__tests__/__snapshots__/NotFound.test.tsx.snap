// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <nav
    class="sc-dlfnbm fKtKuo"
  >
    <h1>
      Invision
    </h1>
  </nav>
  <div
    class="sc-bdfBwQ dMvPSB"
  >
    <header>
      <h1>
        <span>
          404
        </span>
         
        <br />
        Page Not Found
      </h1>
      <p>
        Page in development, come back soon
      </p>
      <button
        class="sc-gsTCUz lhzdGl"
        color="#FFF"
      >
        Go to home
      </button>
    </header>
    <img
      alt="Developing page"
      src="contruction-illustration.svg"
    />
  </div>
</DocumentFragment>
`;
